reV/supply_curve/supply_curve.py

```diff
@@ -256,30 +269,51 @@ def _merge_sc_trans_tables(cls, sc_points, trans_table,
             This is performed by merging left with trans_table, so there may be
             rows with nan sc_gid.
         """
-        trans_table = cls._parse_trans_table(trans_table)
+        if isinstance(trans_table, (list, tuple)):
+            trans_sc_table = []
+            for table in trans_table:
+                trans_sc_table.append(cls._merge_sc_trans_tables(
+                    sc_points, table, sc_cols=sc_cols))
 
-        if isinstance(sc_cols, tuple):
-            sc_cols = list(sc_cols)
+            trans_sc_table = pd.concat(trans_sc_table)
+        else:
+            trans_table = cls._parse_trans_table(trans_table)
+
+            if compare_cap and 'max_cap' in trans_table:
+                trans_sc_table = []
+                for cap, table in trans_table.groupby('max_cap'):
+                    mask = sc_points['capacity'] <= cap
+                    trans_sc_table.append(cls._merge_sc_trans_tables(
+                        sc_points.loc[mask].copy(),
+                        table,
+                        compare_cap=False,
+                        sc_cols=sc_cols))
+
+                trans_sc_table = pd.concat(trans_sc_table)
```

Comment:
regarding syntax and read-ability, I tend to think this spacing method isn't appropriate here. I understand that you want an empty space after for-loops, but having the pd.concat() call directly above the else: statement visually joins the two, while in fact it would be more intuitive to see the pd.concat() grouped with the for-loop instead of the else: indent level. Does that make sense? If you like it this way don't feel obligated to change it, but I just dont believe in "absolute" rules in cases like this.

Reply:
I think thats fair and a good "exception"

Reply:
Is there an easy way to reduce the total amount of indentation? I'm looking at this with a lot of github comments in it, but it's hard to see what lines up with what.

Reply:
I would almost always suggest having the code side by side with the github markup. The github sequential markup (instead of side by side) is really difficult to read. 

As far as reducing indentation: write more functions!

Reply:
And ruin my recursion!?!?!

Reply:
haha no no, just general comment about lots of indentation